src/creds.rs

```diff
@@ -105,30 +106,66 @@ impl Codec for CredentialType {
     // }
 }
 
+/// Struct containing an X509 certificate chain, as per Spec.
 #[derive(Debug, PartialEq, Clone)]
-pub enum Credential {
+pub struct Certificate {
+    cert_data: Vec<u8>,
+}
+
+/// This struct contains the different available certificates.
+#[derive(Debug, PartialEq, Clone)]
+pub enum MLSCredentialType {
     Basic(BasicCredential),
+    X509(Certificate),
+}
+
+/// Struct containing MLS credential data, where the data depends on the type.
+#[derive(Debug, PartialEq, Clone)]
+pub struct Credential {
+    credential: MLSCredentialType,
 }
 
 impl Credential {
+    /// Verify a signature of a given payload against the public key contained
+    /// in a credential.
     pub fn verify(&self, payload: &[u8], signature: &Signature) -> bool {
-        match self {
-            Credential::Basic(basic_credential) => basic_credential.ciphersuite.verify(
+        match &self.credential {
+            MLSCredentialType::Basic(basic_credential) => basic_credential.ciphersuite.verify(
                 signature,
                 &basic_credential.public_key,
                 payload,
             ),
+            // TODO: implement verification for X509 certificates
+            MLSCredentialType::X509(_) => panic!("X509 certificates are not yet implemented."),
+        }
+    }
+    /// Get the identity of a given credential.
+    pub fn get_identity(&self) -> &Vec<u8> {
+        match &self.credential {
+            MLSCredentialType::Basic(basic_credential) => &basic_credential.identity,
+            // TODO: implement getter for identity for X509 certificates
```

Comment:
A TODO should always get an issue referenced if possible. We should have an issue that is for implementing x509 support.